### DISCUSSION TITLE
Change instrument on a track

### DISCUSSION THREAD

**Discussion by y4cer** [mido/mido#396] [Q&A] [answered]
I have initial midi file with its own track and instrument, my goal is to add new `MidiTrack()` with another instrument. The problem is the following: when I add new track with its instrument, the instrument of the old track gets overridden and I obtain the melody played only with one new instument.

```python
mid = MidiFile("input1.mid")
mid.add_track("new_track")

track = mid.tracks[-1]
# here I set the instrument to kalimba
track.append(Message('control_change', control=0, value=0x79, channel=0, time=0))
track.append(Message('control_change', control=32, value=0x00, channel=0, time=0))
track.append(Message('program_change', program=0x6c, channel=0, time=0))
# here goes code with adding notes to new_track
```

**Comment by rdoursenaud** [comment, marked as answer]
Pretty sure your original file already uses the first channel.
If that's the case, it doesn't matter that the data is in a different track it will end up in the same place on your sound generator.
Maybe use a channel that is not already used in the file you're trying to modify?

**Reply by y4cer** [reply]
Thank you indeed!

**Reply by rdoursenaud** [reply]
You're welcome :)
